### ISSUE TITLE
Pango functions are not pure

### ISSUE THREAD

**Issue by tomjaguarpaw** [gtk2hs/gtk2hs#150]
Prelude> import Graphics.Rendering.Pango
    Prelude Graphics.Rendering.Pango> f
    Normal
    Prelude Graphics.Rendering.Pango> fontDescriptionSetStyle f StyleItalic
    Prelude Graphics.Rendering.Pango> f
    Italic

This is astonishing.  Has no one noticed this before?

**Comment by bergey** [comment]
That seems evident from the name & type: `FontDescription -> FontStyle -> IO ()`.  A `FontDescription` is a `ForeignPtr`, and the function certainly isn't returning a new `FontDescription`.

What behavior would you expect instead?

**Comment by tomjaguarpaw** [comment]
I would expect *reading* a `FontDescription` to require `IO` if its value can change!

**Comment by bergey** [comment]
Fair enough.  The `Show` instances use `unsafePerformIO`.  It's hacky, and there's certainly no corresponding `Read` instance.
